core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -335,4 +422,29 @@ public final boolean isTlsEnabled() {
     public final void setTlsEnabled(boolean enabled) {
         this.tlsEnabled = enabled;
     }
+
+    /**
+     * Returns an instance of {@link ProxyOptions} configured with the proxyHost, proxyPort, proxyUsername and
+     * proxyPassword.
+     * <p>
+     * If proxyHost and proxyPort are not set, then {@code null} is returned
+     * </p>
+     * 
+     * @return either an instance of {@link ProxyOptions} if proxy details are provided, {@code null} otherwise.
+     */
+    public ProxyOptions getProxyOptions() {
+        if (!StringUtils.isEmpty(proxyHost) &&
+                isValidPort(proxyPort)) {
+            ProxyOptions proxyOptions = new ProxyOptions().setHost(proxyHost)
```

Comment:
You seem to assume that the client wants to connect via an HTTP proxy, right?
This is probably not, what you usually do when connecting to a TCP socket other than an HTTP port.
FMPOV the configuration needs to take the proxy type (`ProxyOptions.setType()`) into account, using a default of `ProxyType.SOCKS5`).

Reply:
Good point. I will add it.